include/class.ticket.php

```diff
@@ -2314,6 +2314,13 @@ function transfer(TransferForm $form, &$errors, $alert=true) {
         // Log transfer event
         $this->logEvent('transferred');
 
+        if ($this->thread->referrals) {
```

Comment:
This is probably the **most expensive** way of implementing the check - **1)** Checking each referral (regardless of the type of object) **2)** getName() will trigger object lookup for each check!

I recommend the following;

```php

if (($referral=$this->thread->getReferral($dept->getId(), 
       ObjectModel::OBJECT_TYPE_DEPT)))
      $referral->delete();
```

`getReferral()` uses `findFirst(..)` which doesn't hit the database.